superset/security.py

```diff
@@ -52,10 +51,29 @@
     'datasource_access',
     'database_access',
     'muldelete',
+    'all_datasource_access',
 ])
 READ_ONLY_PRODUCT = set(
     product(READ_ONLY_PERMISSION, READ_ONLY_MODELVIEWS))
 
+USER_DEFINED_PERMISSIONS = set([
+    'database_access',
+    'datasource_access',
+    'metric_access',
+])
+
+GLOBAL_VIEW_MENUES = set([
+    'all_datasource_access',
+    'all_database_access',
+])
+
+
+def is_user_defined_permission(perm):
+    if (perm.permission.name in USER_DEFINED_PERMISSIONS and
+            perm.view_menu.name not in GLOBAL_VIEW_MENUES):
```

Comment:
I don't think the overlap exists

`all_database_access` view menu has a matching permission name of `all_database_access`, not `database_access`